realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -672,6 +702,20 @@ Builder schema(Class<? extends RealmModel> firstClass, Class<? extends RealmMode
          * @return the created {@link RealmConfiguration}.
          */
         public RealmConfiguration build() {
+            // Check that readOnly() was applied to legal configuration. Right now it should only be allowed if
+            // an assetFile is configured
+            if (readOnly) {
+                if (initialDataTransaction != null) {
+                    throw new IllegalStateException("This Realm is marked as read-only. Read-only Realms cannot use initialData(Realm.Transaction).");
+                }
+                if (assetFilePath == null) {
+                    throw new IllegalStateException("Only Realms provided using 'assetFile(path)' can be marked read-only. No such Realm was provided.");
+                }
+                if (deleteRealmIfMigrationNeeded) {
+                    throw new IllegalStateException("'deleteRealmIfMigrationNeeded()' and read-only Realms cannot be combined");
```

Comment:
Exactly.
Suppose an app contains an master data as an read-only Realm database.
If the new version of the app need to use a master with updated schema,
I think the app need to delete the old one and then copy the new Realm file 
from assets.

`deleteIfMigrationNeeded()` helps those scenario much.